test/services/school/index.test.js

```diff
@@ -41,7 +42,51 @@ describe('school service', () => {
 			sampleSchoolData = await School.findById(school._id).lean().exec();
 			delete sampleSchoolData._id;
 		});
+		it('should be possible to see only limited amount of fields if the user is not logged in', async () => {
+			const params = {
+				provider: 'rest',
+				headers: {
+					authorization: undefined
+				},
+				account: undefined,
+			}
+			const result = await schoolService.find(params);
+			const expectedFields = ['purpose', 'name', 'id', 'years', 'isTeamCreationByStudentsEnabled'];
+			const notExpectedFields = ['fileStorageType', 'documentBaseDir', 'inMaintenance', 'currentYear', 'systems', 'federalState'];
+			result.data.forEach((school) => {
+				expectedFields.forEach(field => {
+					expect(school).to.haveOwnProperty(field);
+				});
+				notExpectedFields.forEach(field => {
+					expect(school).to.not.haveOwnProperty(field);
+				})
+			})
+		});
+		it('should be possible to see all fields if the call is done from the server', async () => {
+			const params = { provider: undefined }
+
+			const result = await schoolService.find(params);
+			const expectedFields = [
+				'purpose',
+				'name',
+				'id',
+				'years',
+				'isTeamCreationByStudentsEnabled',
+				'fileStorageType',
+				'documentBaseDir',
+				'inMaintenance',
+				'currentYear',
+				'systems',
+				// 'federalState', // there is a school in the test database that doesn't have this field
+			];
 
+			result.data.forEach((school) => {
+				expectedFields.forEach(field => {
+					expect(school).to.haveOwnProperty(field)
+				}
+			)
```

Comment:
Codacy found an issue: [Expected indentation of 4 tabs but found 3.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=5734878)